src/parser/Parser.ts

```diff
@@ -523,19 +525,19 @@ export class Parser {
     }
 
     private enumDeclaration(): EnumStatement {
-        const result = new EnumStatement({ name: {} as any, body: [] });
-        this.warnIfNotBrighterScriptMode('enum declarations');
-
-        const parentAnnotations = this.enterAnnotationBlock();
-
-        result.tokens.enum = this.consume(
+        const enumToken = this.consume(
             DiagnosticMessages.expectedKeyword(TokenKind.Enum),
             TokenKind.Enum
         );
+        const nameToken = this.tryIdentifier(...this.allowedLocalIdentifiers);
 
-        result.tokens.name = this.tryIdentifier(...this.allowedLocalIdentifiers);
+        this.warnIfNotBrighterScriptMode('enum declarations');
+
+        const parentAnnotations = this.enterAnnotationBlock();
 
         this.consumeStatementSeparators();
+
+        const body: (EnumMemberStatement | CommentStatement)[] = [];
```

Comment:
The grouping syntax here resembles arrow functions so I got a little confused until I read the whole line. Maybe use the `Array` type instead?
```suggestion
        const body: Array<EnumMemberStatement | CommentStatement> = [];
```